xbmc/interfaces/json-rpc/FileOperations.cpp

```diff
@@ -213,6 +214,50 @@ JSONRPC_STATUS CFileOperations::GetFileDetails(const std::string &method, ITrans
   return OK;
 }
 
+JSONRPC_STATUS CFileOperations::SetFileDetails(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  std::string file = parameterObject["file"].asString();
+  if (!CFile::Exists(file))
+    return InvalidParams;
+
+  if (!CFileUtils::RemoteAccessAllowed(file))
+    return InvalidParams;
+
+  std::string media = parameterObject["media"].asString();
+  StringUtils::ToLower(media);
+
+  if (media.compare("video") != 0)
```

Comment:
I would do this check at the very beginning because it's a lot cheaper than checking if the file exists and is allowed.